handler/handler.go

```diff
@@ -102,29 +116,47 @@ func (h *Handler) AddAnnotation(w http.ResponseWriter, r *http.Request) {
 	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
 
 	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+	schemaVersion := r.Header.Get(annotations.SchemaVersionHeader)
+	if schemaVersion == "" {
+		schemaVersion = annotations.DefaultSchemaVersion
+	}
+	ctx = context.WithValue(ctx, annotations.SchemaVersionHeaderKey(annotations.SchemaVersionHeader), schemaVersion)
+
+	origin := r.Header.Get(annotations.OriginSystemIDHeader)
+	if origin == "" {
+		handleWriteErrors("Invalid request", errors.New("X-Origin-System-Id header missing"), writeLog, w, http.StatusBadRequest)
+		return
+	}
+	ctx = context.WithValue(ctx, annotations.OriginSystemIDHeaderKey(annotations.OriginSystemIDHeader), origin)
 
-	addedAnnotation := annotations.Annotation{}
-	err := json.NewDecoder(r.Body).Decode(&addedAnnotation)
+	var addedAnnotationBody map[string]interface{}
+	err := json.NewDecoder(r.Body).Decode(&addedAnnotationBody)
 	if err != nil {
 		handleWriteErrors("Error decoding request body", err, writeLog, w, http.StatusBadRequest)
 		return
 	}
 
-	if !mapper.IsValidPACPredicate(addedAnnotation.Predicate) {
+	addedAnnotation := addedAnnotationBody["annotation"].(map[string]interface{})
```

Comment:
`question`: Throughout the code there are many places where we are using type assertion without checking if it was successful, we are doing this because we are aware of what is contained in the interface, however this makes the program less panic proof. Do you think we should handle type assertion checks throughout the code?

Reply:
Yes you are correct. In theory we should handle type assertion checks throughout the code. However with the next PR we would place validation on the incoming requests, which should minimise the risk of panics. Therefore I have decided not to do so many type assertion checks to make the code more readable/clean.